libs/core/src/lib/select/option/option.component.ts

```diff
@@ -61,81 +104,150 @@ export class OptionComponent implements OnInit, OnDestroy {
+     */
+    get viewValue(): string {
         return (this._elementRef.nativeElement.textContent || '').trim();
     }
 
-    /** Set control selected state
-     * @param value - whether is selected
-     * @param controlChange - whether is a change initially coming from OptionComponent
-     * */
-    setSelected(value: boolean, controlChange: boolean): void {
-        this.selected = value;
+    /**
+     * Whether or not the option is currently active and ready to be selected.
+     * An active option displays styles as if it is focused, but the
+     * focus is actually retained somewhere else. This comes in handy
+     * for components like autocomplete where focus must remain on the input.
+     */
+    get active(): boolean {
+        return this._active;
+    }
 
-        if (value) {
-            this._selectProxy.optionStateChange$.next({option: this, controlChange: controlChange})
+    /**
+     *  Since parent components could be using the option's label to display the selected values
+     *  and they don't have a way of knowing if the option's label has changed
+     * we have to check for changes in the DOM ourselves and dispatch an event. These checks are
+     * relatively cheap, however we still limit them only to selected options in order to avoid
+     * hitting the DOM too often.
+     *
+     * @hidden
+     */
+    ngAfterViewChecked(): void {
+
+        if (this.selected) {
+            const viewValue = this.viewValue;
+
+            if (viewValue !== this._mostRecentViewValue) {
+                this._mostRecentViewValue = viewValue;
+                this._stateChanges.next();
+            }
         }
-        this._changeDetRef.markForCheck();
     }
 
     /** Focuses the element. */
-    focus(): void {
-        (this._elementRef.nativeElement as HTMLElement).focus();
+    focus(_origin?: FocusOrigin, options?: FocusOptions): void {
+        const element = this._elementRef.nativeElement;
+
+        if (typeof element.focus === 'function') {
+            element.focus(options);
+        }
     }
 
     /** Returns HTMLElement representation of the component. */
     getHtmlElement(): HTMLElement {
         return this._elementRef.nativeElement as HTMLElement;
     }
 
+
+    /** Selects the option. */
+    select(): void {
+        if (!this.selected) {
+            this.selected = true;
+            this._changeDetectorRef.markForCheck();
+            this.selectionChange.emit(new FdOptionSelectionChange(this, false));
+        }
+    }
+
+    /** Deselects the option. */
+    deselect(): void {
+        if (this.selected) {
+            this.selected = false;
+            this._changeDetectorRef.markForCheck();
+            this.selectionChange.emit(new FdOptionSelectionChange(this, false));
+        }
+    }
+
+
+    getLabel(): string {
+        return this.viewValue;
+    }
+
+
+    /** Ensures the option is selected when activated from the keyboard. */
+    _handleKeydown(event: KeyboardEvent): void {
```

Comment:
What about the `_` prefix? We decided to reserve it only for private properties/methods

Reply:
@JKMarkowski  This just means that this method is not public and its more private API but since you want this to be called from outside, or template, its hybrid. `_` you tell its private API but you make it public as you want to access it. 

This is common practice. also in other opensource project, Angular.io, material.

Reply:
But In this specifc case it makes sense as we dont reference templates

Reply:
All updated to much our guidelines,

Reply:
Did you guys agreed to stick to the guidelines (`_` prefix only for private properties) or we allow to use it for internal public properties which should not be used by the user?
Asking because `_handleKeydown` still has prefix.

Reply:
@JKMarkowski @salarenko this one seems like leftover, I replaced all the public that are shareble and those which does kept private. not this one.